lib/kubernetes-deploy/runner.rb

```diff
@@ -112,6 +112,21 @@ def template_variables
 
     private
 
+    # Inspect the file that was mentioned in the kubectl output
+    # to make it easier for developer to understand what's going on
+    def inspect_kubectl_out_for_yaml(out)
+      # Output example:
+      # error: unable to decode "/tmp/path/to/file": [pos 96]: json: expect char '"' but got char '{'
+      if match = out.match(/error:(.+)"(?<path>\S+)":+/)
```

Comment:
> Could we have a regex that'll look for things that look more like files?

I don't want to fix to `/tmp/*` as a pattern of temp file since the temporary directory may vary:

```
Tempfile.new.path
=> "/var/folders/9k/zxhywlyx2nx_3z3qnrzpkjd00000gn/T/20170227-55216-1goaa6q"
``` 

Any anyway we check if it's a file below.

In this case we can only inspect `error: unable to decode` issues. It's the most common scenario that happens when the YAML has syntax errors that can't be caught by `--dry-run` (https://github.com/kubernetes/kubernetes/issues/42057)